app/src/main/java/com/github/brandon_patterson/chemist/matter/Element.java

```diff
@@ -0,0 +1,192 @@
+    CD(R.string.element_name_048, 48),
+    IN(R.string.element_name_049, 49),
+    SN(R.string.element_name_050, 50),
+    SB(R.string.element_name_051, 51),
+    TE(R.string.element_name_052, 52),
+    I(R.string.element_name_053, 53),
+    XE(R.string.element_name_054, 54),
+    CS(R.string.element_name_055, 55),
+    BA(R.string.element_name_056, 56),
+    LA(R.string.element_name_057, 57),
+    CE(R.string.element_name_058, 58),
+    PR(R.string.element_name_059, 59),
+    ND(R.string.element_name_060, 60),
+    PM(R.string.element_name_061, 61),
+    SM(R.string.element_name_062, 62),
+    EU(R.string.element_name_063, 63),
+    GD(R.string.element_name_064, 64),
+    TB(R.string.element_name_065, 65),
+    DY(R.string.element_name_066, 66),
+    HO(R.string.element_name_067, 67),
+    ER(R.string.element_name_068, 68),
+    TM(R.string.element_name_069, 69),
+    YB(R.string.element_name_070, 70),
+    LU(R.string.element_name_071, 71),
+    HF(R.string.element_name_072, 72),
+    TA(R.string.element_name_073, 73),
+    W(R.string.element_name_074, 74),
+    RE(R.string.element_name_075, 75),
+    OS(R.string.element_name_076, 76),
+    IR(R.string.element_name_077, 77),
+    PT(R.string.element_name_078, 78),
+    AU(R.string.element_name_079, 79),
+    HG(R.string.element_name_080, 80),
+    TL(R.string.element_name_081, 81),
+    PB(R.string.element_name_082, 82),
+    BI(R.string.element_name_083, 83),
+    PO(R.string.element_name_084, 84),
+    AT(R.string.element_name_085, 85),
+    RN(R.string.element_name_086, 86),
+    FR(R.string.element_name_087, 87),
+    RA(R.string.element_name_088, 88),
+    AC(R.string.element_name_089, 89),
+    TH(R.string.element_name_090, 90),
+    PA(R.string.element_name_091, 91),
+    U(R.string.element_name_092, 92),
+    NP(R.string.element_name_093, 93),
+    PU(R.string.element_name_094, 94),
+    AM(R.string.element_name_095, 95),
+    CM(R.string.element_name_096, 96),
+    BK(R.string.element_name_097, 97),
+    CF(R.string.element_name_098, 98),
+    ES(R.string.element_name_099, 99),
+    FM(R.string.element_name_100, 100),
+    MD(R.string.element_name_101, 101),
+    NO(R.string.element_name_102, 102),
+    LR(R.string.element_name_103, 103),
+    RF(R.string.element_name_104, 104),
+    DB(R.string.element_name_105, 105),
+    SG(R.string.element_name_106, 106),
+    BH(R.string.element_name_107, 107),
+    HS(R.string.element_name_108, 108),
+    MT(R.string.element_name_109, 109),
+    DS(R.string.element_name_110, 110),
+    RG(R.string.element_name_111, 111),
+    CN(R.string.element_name_112, 112),
+    NH(R.string.element_name_113, 113),
+    FL(R.string.element_name_114, 114),
+    MC(R.string.element_name_115, 115),
+    LV(R.string.element_name_116, 116),
+    TS(R.string.element_name_117, 117),
+    OG(R.string.element_name_118, 118);
+
+    private int name_id;
+    private int atomicNumber;
+
+    Element(int name_id, int atomicNumber)
+    {
+        this.name_id = name_id;
```

Comment:
Where does `name_id` come from? Is this from the enum? If so, do the `name_id` and `atomicNumber` differ at all in this case?

Reply:
`name_id` is just an id that maps to a resource, e.g. `R.string.element_name_01` above. `name_id` and `atomicNumber` come from lines 8-125, where I declare the enum values.

Check out the [docs](https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html) for enums to see about using constructors. That functionality doesn't exist in a lot of other languages, and it was new to me here.

Technically I think we could get away with implicit atomicNumbers, since enums are just ints counting up from 0. We'd just need to add a ~`NONE` element first. That feels like an implementation detail though, so I feel safer not depending on the implicit id.

(Just noticed `name_id` is accidentally snake case, will switch it to lowerCamel.)